src/main/scala/org/broadinstitute/hail/driver/ExportVCF.scala

```diff
@@ -207,7 +207,28 @@ object ExportVCF extends Command {
           sb.append("GT:AD:DP:GQ:PL")
         gs.foreach { g =>
           sb += '\t'
-          sb.append(g)
+
+          sb.append(g.gt.map { gt =>
+            val p = Genotype.gtPair(gt)
+            s"${ p.j }/${ p.k }"
+          }.getOrElse("./."))
+          sb += ':'
+          sb.append(g.ad.map(_.mkString(",")).getOrElse("."))
+          sb += ':'
+          sb.append(g.dp.map(_.toString).getOrElse("."))
+          sb += ':'
+          sb.append(g.gq.map(_.toString).getOrElse("."))
+          sb += ':'
+          if (!g.isDosage)
+            sb.append(g.pl.map(_.mkString(",")).getOrElse("."))
+          else
+            sb.append(g.dosage.map(_.mkString(",")).getOrElse("."))
+
+          val regex = """(?::\.)+$""".r
+          val regexMatch = regex.findFirstMatchIn(sb)
+          regexMatch.foreach { m =>
+            sb.delete(m.start, m.end)
+          }
```

Comment:
The regex is now quadratic in the number of genotypes, since the genotypes are added one at a time to `sb` but the regex scans the entire `StringBuffer`.  (It would be a cute optimization compile regular expressions that end in `$` backwards, but Java/Scala doesn't do that.  I tried your regex against successively larger strings using `printTime` from our Utils and it definitely scales linearly in the length of the string.)  With this approach, I think you have to emit only the necessary fields.

Maybe something like this?

```
  (g.ad.map(_.mkString(",")), 
    g.dp.map(_.toString),
    g.gq.map(_.toString), 
    if (g.isDosage)
      g.dosage.map(_.mkString(","))
    else
      g.pl.map(_.mkString(","))) match {
    case (None, None, None, None) =>
    case (Some(ad), None, None, None) =>
      sb.append(ad)
    case (ad, Some(dp), None, None) =>
      etc...
  }
```

Otherwise, looks great!